defender/wallet-migrator/autotask-1/index.js

```diff
@@ -0,0 +1,173 @@
+  if (covalentApiKey === undefined) {
+    throw new Error('covalent-ap-key must be defined in config.dev.yml file or Defender->AutoTask->Secrets');
+  }
+
+  const network = secrets[networkSecretName];
+
+  // ensure that the network exists
+  if (network === undefined) {
+    throw new Error('network must be defined in config.dev.yml file or Defender->AutoTask->Secrets');
+  }
+
+  // get chain id depending on network
+  let chainId;
+  if (network === 'goerli') {
+    chainId = '5';
+  } else if (network === 'mainnet') {
+    chainId = '1';
+  }
+
+  // query senders wallet balance for all erc20 and erc721 tokens
+  const method = 'get';
+  const headers = {
+    'Content-Type': 'application/json',
+  };
+  const url = `https://api.covalenthq.com/v1/${chainId}/address/${senderWalletAddress}/balances_v2/?no-nft-fetch=true&nft=true`;
+  const auth = {
+    username: covalentApiKey,
+  };
+
+  let responseData;
+
+  // get balance of all tokens sender address owns
+  try {
+    let response = await get(url, method, headers, auth);
+    responseData = response.data.data.items;
+  } catch (error) {
+    console.log('error retrieving wallet balance:');
+    console.log(error);
+  }
+
+  // create new provider and signer to perform on-chain transfers
+  console.debug('Creating DefenderRelayProvider');
+  const provider = new DefenderRelayProvider(autotaskEvent);
+  console.debug('Creating DefenderRelaySigner');
+  const signer = new DefenderRelaySigner(autotaskEvent, provider, { speed: 'fast' });
+  const relayerAddress = await signer.getAddress();
+
+  let erc20Contract;
+  let erc721Contract;
+
+  // these initial contract addresses are just placeholders - they will be replaced by each erc20 and erc721 contract address the user holds
+  // it can cause an error with ethers.js if the contract address is invalid, so we are passing in known valid contract addresses
+  // as place holders for erc20 and erc721 tokens on both goerli and mainnet
+  if (network === 'goerli') {
+    erc20Contract = new ethers.Contract('0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6', erc20Abi, signer);
+    erc721Contract = new ethers.Contract('0xCfD7cEF761A60dFBA0D240ee4fF82f7f51242675', erc721Abi, signer);
+  } else if (network === 'mainnet') {
+    erc20Contract = new ethers.Contract('0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2', erc20Abi, signer);
+    erc721Contract = new ethers.Contract('0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D', erc721Abi, signer);
+  }
+
+  for (var i = 0; i < responseData.length; i++) {
+    const item = responseData[i];
+    const balance = item.balance;
+    const isNft = item.type === 'nft';
+    const isDust = item.type === 'dust'; // tokens with less than $0.1 in spot fiat value get classified as dust (ignore unless it's eth)
+    const decimals = item.contract_decimals;
+    const symbol = item.contract_ticker_symbol;
+    const isEther = item.contract_address === '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
```

Comment:
```suggestion
    const {
      balance, type,
      contract_decimals: decimals,
      contract_ticker_symbol: symbol,
      contract_address: contractAddress,
    } = responseData[i];
    const isNft = type === 'nft';
    const isDust = type === 'dust'; // tokens with less than $0.1 in spot fiat value get classified as dust (ignore unless it's eth)
    const isEther = contractAddress === '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
```
Destructure like in `approveTokens`